hail/src/main/scala/is/hail/utils/TextTableReader.scala

```diff
@@ -163,7 +165,8 @@ object TextTableReader {
       float64Matcher)
     val nMatchers = matchers.length
 
-    val imputation = values.mapPartitions { it =>
+    val (imputation, allDefined) = values.mapPartitions { it =>
+      val allDefined = Array.fill(nFields)(false)
```

Comment:
I think this should be (true), below this allDefined is negated when a field is in the missing.

Also, if missing always corresponds to fields in 0 .. nFields, allDefined could just be missing.size == 0

Reply:
yeah, you're right, I had it as `anyNA` before and flipped it